jvm-runtime/ftl-runtime/common/deployment/src/main/java/xyz/block/ftl/deployment/TopicsProcessor.java

```diff
@@ -69,7 +71,10 @@ TopicsBuildItem handleTopics(CombinedIndexBuildItem index, BuildProducer<Generat
                                 + " with a concrete type parameter " + iface.name() + " does not extend this interface");
             }
             var partitionMapperClass = partitionMapperType.name();
-            beans.produce(AdditionalBeanBuildItem.unremovableOf(partitionMapperClass.toString()));
+            if (partitionMapperClass.equals(FTLDotNames.SINGLE_PARTITION_MAPPER)) {
```

Comment:
This should be `PARTITION_MAPPER` (and the DotNames class updated) assuming you want to make this the default. Otherwise this line does not do anything.

Reply:
Ahhhh. Removing, as I didn't mean to try to make it the default